### PR TITLE
Ten gateway use http instead of ws

### PR DESCRIPTION
**Pull request by zkokelj**
### Why this change is needed

We want to use http connections from the gateway to the validator. Having web socket connections open for a lot of users can cause gateway to crash. It is also problematic if we want to have it behind load balancer.
Websocket connections are still needed for subscriptions.

https://github.com/ten-protocol/ten-internal/issues/2505

### What changes were made as part of this PR

Changed ws connection to http, creating ws connections for subscriptions + added a test for subscription topics

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/ten-protocol/ten-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [x] PR checks reviewed and performed 

https://github.com/ten-protocol/ten-test/actions/runs/7504133752/job/20430444139

### CODE REVIEW

File: integration/obscurogateway/tengateway_test.go [hunk 1]
```diff
@@ -244,6 +245,83 @@
...
+
+	// wait a few seconds to be completely sure all events arrived
+	time.Sleep(time.Second * 3)
+
+	// Assert the number of logs received by each client
+	// user0 should see two lifecycle events (1 for each interaction with the smart contract)
+	assert.Equal(t, 2, len(user0logs))
+	// user1 should see only one event (the other is filtered out because of the topic filter)
+	assert.Equal(t, 1, len(user1logs))
+}
```
[the remark this answers is not in the document]
Comment (zkokelj) [reply integration/obscurogateway/tengateway_test.go]: fixed a todo

File: integration/obscurogateway/tengateway_test.go [hunk 2]
```diff
@@ -90,12 +90,13 @@ func TestTenGateway(t *testing.T) {
...
 	// run the tests against the exis
 	for name, test := range map[string]func(*testing.T, string, string, wallet.Wallet){
-		//"testAreTxsMinted":            testAreTxsMinted, this breaks the other tests bc, enable once concurency issues are fixed
+		//"testAreTxsMinted":            testAreTxsMinted, this breaks the other tests bc, enable once concurrency issues are fixed
 		"testErrorHandling":                    testErrorHandling,
 		"testMultipleAccountsSubscription":     testMultipleAccountsSubscription,
 		"testErrorsRevertedArePassed":          testErrorsRevertedArePassed,
 		"testUnsubscribe":                      testUnsubscribe,
 		"testClosingConnectionWhileSubscribed": testClosingConnectionWhileSubscribed,
+		"testSubscriptionTopics":               testSubscriptionTopics,
```
[the remark this answers is not in the document]
Comment (zkokelj) [reply integration/obscurogateway/tengateway_test.go:99]: not in scope of this task

File: tools/walletextension/accountmanager/account_manager.go [hunk 3]
```diff
@@ -103,29 +111,51 @@ func (m *AccountManager) ProxyRequest(rpcReq *wecommon.RPCRequest, rpcResp *inte
...
-	if len(rpcReq.Params) < 2 {
-		return clients, nil
+	if len(rpcReq.Params) > 1 {
+		filteredAccounts, err := m.filterAccounts(rpcReq, accounts)
+		if err != nil {
+			return nil, err
+		}
+		// return filtered clients if we found any
+		if len(filteredAccounts) > 0 {
+			return m.createClientsForAccounts(filteredAccounts, userPrivateKey)
```
Comment (otherview) [inline tools/walletextension/accountmanager/account_manager.go:145]: Suggestion: 

```suggestion
			accounts = filteredAccounts
```

**Review by otherview — APPROVED** [review]
Minor suggestion, non-blocking tho.

**Pull request merged**